### PR TITLE
handle-config-via-environment

### PR DESCRIPTION
**Pull request by PierreSavatte**
Address the issue #2

### CODE REVIEW

File: kirby/api/context.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import os
+from smart_getenv import getenv
+import json
+from copy import deepcopy
+
+TYPES = {"str": str, "list": list}
```
Comment (ericgazoni) [inline kirby/api/context.py]: I get that this is required for serialization/deserialization, but `pickle` would handle this in a more "standard" way

File: kirby/api/context.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+KIRBY_ENV_SIGNATURE = "KIRBY_ENV_SIGNATURE"
```
Comment (ericgazoni) [inline kirby/api/context.py]: just for the sake of it, maybe add a couple of underscore in front of the name, just to make clear this is not something to be messed with

File: kirby/api/context.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    return config
+
+
+class ContextManager:
+    def __init__(self, config):
+        self.config = config
+        return
+
+    def load(self):
+        encoder = json.JSONEncoder()
```
Comment (ericgazoni) [inline kirby/api/context.py]: `pickle` would handle this for you

File: kirby/api/context.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        os.environ[KIRBY_ENV_SIGNATURE] = encoder.encode(
+            parse_types(self.config)
+        )
+
+
+class Context:
+    def get_signature(self):
+        decoder = json.JSONDecoder()
```
Comment (ericgazoni) [inline kirby/api/context.py]: `pickle` would handle this for you

File: kirby/api/context.py (hunk 1 above)
Comment (PierreSavatte) [reply kirby/api/context.py]: Now using pickle

File: kirby/api/context.py (hunk 3 above)
Comment (PierreSavatte) [reply kirby/api/context.py]: Now using pickle

File: kirby/api/context.py (hunk 4 above)
Comment (PierreSavatte) [reply kirby/api/context.py]: Now using pickle

File: kirby/api/context.py (hunk 2 above)
Comment (PierreSavatte) [reply kirby/api/context.py]: Added `__` in front of it

**Pull request merged**